src/openzaak/utils/db.py

```diff
@@ -0,0 +1,17 @@
+# SPDX-License-Identifier: EUPL-1.2
+# Copyright (C) 2022 Open Zaak maintainers
+import zlib
+from contextlib import contextmanager
+
+from django.db import connections, transaction
+
+
+@contextmanager
+def pg_advisory_lock(lock_id: str, using="default"):
+    _lock_id = zlib.crc32(lock_id.encode("utf-8"))
+    with transaction.atomic(using=using):
+        connection = connections[using]
+        with connection.cursor() as cursor:
+            sql = f"SELECT pg_advisory_xact_lock({_lock_id})"
```

Comment:
Is it correct that this lock is not on the table level, but on the application level? 
Why don't you want to lock specific table (ZaakIdentificatie) in exclusive mode here?

Reply:
correct, it's application specific and there is no actual DB locking.

Using an exclusive lock would lock the table for the duration of the entire request and thereby also block read operations to this table. This would mean that for the duration of zaak identificatie generation, it is not possible to query a list of zaken or detail of zaken, or even any resource with a FK to zaak that hits the `zaken_zaak` table, as the concrete inheritance implies an INNER JOIN on the `zaken_zaakidentificatie` table.

So this would seriously hurt concurrency and actively make things worse than they currently are as it also affects reading. Pessimistic locking is something I wanted to avoid as much as possible.